lib/syscall_shim/uk_syscall_binary.c

```diff
@@ -112,17 +112,20 @@ void ukplat_syscall_handler(struct __regs *r)
 #endif /* CONFIG_LIBSYSCALL_SHIM_STRACE */
 
 #if CONFIG_LIBSYSCALL_SHIM_HANDLER_ULTLS
-	uk_thread_uktls_var(self, _uk_syscall_ultlsp) = 0x0;
-
 	/* Restore original TLS only if it was _NOT_
 	 * changed by the system call handler
 	 */
-	if (likely(ukplat_tlsp_get() == self->uktlsp)) {
-		ukplat_tlsp_set(orig_tlsp);
-	} else {
+#if CONFIG_LIBUKDEBUG_PRINTD
+	if (_uk_syscall_ultlsp != ultlsp) {
 		uk_pr_debug("System call updated userland TLS pointer register to %p (before: %p)\n",
-			    (void *) orig_tlsp, (void *) ukplat_tlsp_get());
+			    (void *)ultlsp, (void *)ukplat_tlsp_get());
 	}
+#endif
+
+	ultlsp = _uk_syscall_ultlsp;
```

Comment:
And as `ultlsp` is not used beyond this point, this would be more readable as:
```diff
-	ultlsp = _uk_syscall_ultlsp;
-	uk_thread_uktls_var(self, _uk_syscall_ultlsp) = 0x0;
-
-	ukplat_tlsp_set(ultlsp);
+	ukplat_tlsp_set(_uk_syscall_ultlsp);
+ 	uk_thread_uktls_var(self, _uk_syscall_ultlsp) = 0x0;
```